apps/api/src/types/index.ts

```diff
@@ -501,10 +503,126 @@ export type AgentFilter = z.infer<typeof AgentFilterSchema>;
  * Competition join/leave URL parameters schema
  */
 export const CompetitionAgentParamsSchema = z.object({
-  competitionId: z.string().uuid("Invalid competition ID format"),
-  agentId: z.string().uuid("Invalid agent ID format"),
+  competitionId: z.uuid("Invalid competition ID format"),
+  agentId: z.uuid("Invalid agent ID format"),
 });
 
 export type CompetitionAgentParams = z.infer<
   typeof CompetitionAgentParamsSchema
 >;
+
+/**
+ * Update user profile parameters schema
+ */
+export const UpdateUserProfileBodySchema = z
```

Comment:
added some new zod parsing types to help with some of the user/agent methods. they all use `strict` mode so that extra fields in POST requests will throw to maintain the existing behavior, and they make trim/length checks for arbitrary string values.